spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1932,6 +1967,18 @@ private void doInvokeWithRecords(final ConsumerRecords<K, V> records) {
 			}
 		}
 
+		private ConsumerRecords<K, V> checkEarlyIntercept(ConsumerRecords<K, V> nextArg) {
+			ConsumerRecords<K, V> next = nextArg;
+			if (this.earlyBatchInterceptor != null) {
+				next = this.earlyBatchInterceptor.intercept(next);
+				if (next == null && this.logger.isDebugEnabled()) {
+					this.logger.debug("RecordInterceptor returned null, skipping: "
```

Comment:
This is already a `LogAccessor`, so no need in that `isDebugEnabled()` and probably this log message should go as a `Supplier`.